nameko_grpc/connection.py

```diff
@@ -269,6 +322,10 @@ def send_data(self, stream_id):
             # has been completely sent
             return
 
+        # No headers are present on close so send streams fail to exhaust without
+        # a manual call to flush the queue (will either be an error or END_STREAM stuck)
+        send_stream.flush_queue_to_buffer()
```

Comment:
Nice find. Can you add a regression test case for this?

Reply:
Will have a look. It came up through existing tests so I'll work out which ones cover it.

Reply:
Hmm, not easily. I added it as the tests were timing out (hanging) without it but removing it now and they appear to be passing still. I think it still makes sense to have it but it's tricky to test specifically.